### DISCUSSION TITLE
Adding files to the search path

### DISCUSSION THREAD

**Discussion by marquezj** [mctools/ncrystal#203] [Q&A] [answered]
Quick question: how do I add a search path for ncmat files that can be found by the hooks?

I tried with `NC.addCustomSearchDirectory()`, but the OpenMC hook does not find the files. In the past I copied files into `os.environ['NCRYSTAL_DATADIR']`, but my understanding is that now the standard library is embedded and that variable is undefined.

**Comment by tkittel** [comment, marked as answer]
The short answer is that you simply add the directory to the `NCRYSTAL_DATA_PATH` environment variable. This variable works like any PATH-like environment variable, so multiple directories can be separated by colons (`:`).

The `NCRYSTAL_DATADIR` variable should not be used for such user data - it exists solely as a rarely needed workaround for weird installations that don't know their _standard_ library location (or for when running unit tests out of the CMake build area and the standard library is _not_ embedded). It might be good to rename this variable at some point, to make the usage more clear.

The reason paths added via `NC.addCustomSearchDirectory()` does not propagate to the openmc run, is that it most likely it is running as a subprocesses, which means it will not share memory space and loaded libraries with your python process (so libNCrystal.so is freshly loaded in the openmc process and can't know anything about any past calls you made). The reason that modifying `os.environ['NCRYSTAL_DATA_PATH']` is likely to work, is that environment variables _are_ inherited by daughter processes.

For reference, the NCRYSTAL_DATA_PATH variable is also mentioned in the notebook at https://github.com/mctools/ncrystal-notebooks/blob/main/notebooks/ncrystal1_basic_02_Data_Infrastructure_and_StdDataLib.ipynb and in the "Using NCrystal" wiki page at: https://github.com/mctools/ncrystal/wiki/Using-NCrystal . But clearly, the bright future with a proper NCrystal website will document this better :-)

(also in principle a duplicate of #82)

**Reply by marquezj** [reply]
Perfect. Thanks!
